### DISCUSSION TITLE
编译报错：“appendixes环境未定义”

### DISCUSSION THREAD

**Discussion by WXiaoyao** [note286/xduts#43] [Q&A] [answered]
按`README.md`的步骤下载并编译，使用`xdutspgthesis`，运行时其他环境都可以用，就唯独显示找不到`appendixes`环境：

![image](https://user-images.githubusercontent.com/28039355/218677147-cff5f27c-c12c-4f6a-af8c-e13551cd0f22.png)

请问是我有哪个步骤没做吗？

**Comment by note286** [comment, marked as answer]
XDUTS目前没有这个环境，请阅读最新版PDF手册，Ctrl+F`附录`来了解如何设置附录。

**Comment by note286** [comment]
```latex
\documentclass{xdupgthesis}
\xdusetup{info/appendix=appendix.tex}
\begin{document}
\chapter{欢迎}
使用\LaTeX{}！
\end{document}
```

**Reply by WXiaoyao** [reply]
我确实是拿着旧版本的使用手册、代码风格，用着新版本的cls文件，怪不得没这个环境哈哈。
不过我按照新版本的appendix的使用方法尝试去插入附录，依然失败了，没有生成附录页，麻烦您再帮忙看下原因：
![image](https://user-images.githubusercontent.com/28039355/218684836-867ddc67-1efc-4fd1-b83c-c5dbd265cd6b.png)
<img width="851" alt="1676364488575" src="https://user-images.githubusercontent.com/28039355/218684958-9a73f67a-2854-4de4-9a65-f6058dff9810.png">

**Reply by note286** [reply]
你需要在`appendix.tex`里写章标题啊，`\chapter{一个附录}`。

**Reply by note286** [reply]
另外，看样子你本地是有LaTeX环境的，为啥不本地写，Overleaf写大块头的学位论文不是很方便。

**Reply by note286** [reply]
/(ㄒoㄒ)/~~ star都不给一个

**Reply by WXiaoyao** [reply]
给给给，我是因为刚听了你的建议没来得及回复，忙着在用本地texstudio调试呢。非常感谢，我调试好了就跟我的同学们去推广一波你这个项目，给你点赞~

**Reply by note286** [reply]
哈哈哈，有任何建议都尽管提。

XDUTS提供了很多接口，建议都试试，很多默认值只是为了避免很多新用户第一次无法编译成功，并不是默认值就是最优值。

**Reply by WXiaoyao** [reply]
哈哈哈好嘞，我去研究研究，非常感谢！

**Comment by note286** [comment]
此外，`\frontmatter`、`\mainmatter`和`\backmatter`均不需要。
